src/libs/Middleware/Logging.ts

```diff
@@ -1,30 +1,26 @@
-import _ from 'underscore';
-import lodashGet from 'lodash/get';
 import Log from '../Log';
 import CONST from '../../CONST';
+import Request from '../../types/onyx/Request';
+import Response from '../../types/onyx/Response';
+import Middleware from './types';
 
-/**
- * @param {String} message
- * @param {Object} request
- * @param {Object} [response]
- */
-function logRequestDetails(message, request, response = {}) {
+function logRequestDetails(message: string, request: Request, response?: Response | void) {
     // Don't log about log or else we'd cause an infinite loop
     if (request.command === 'Log') {
         return;
     }
 
-    const logParams = {
+    const logParams: Record<string, unknown> = {
         command: request.command,
         shouldUseSecure: request.shouldUseSecure,
     };
 
-    const returnValueList = lodashGet(request, 'data.returnValueList');
+    const returnValueList = request?.data?.returnValueList;
```

Comment:
We like to be careful in these situations, especially since lodash previously handled things gracefully. We're adding extra ?. for safety until all components, tests, etc., are written in JavaScript.

Reply:
I don't get it, if we are defining that `request` is of type `Request` above, doesn't that ensure that it won't be `null` (or any other type)?
Also, how/when will we remove all the unneeded safety checks?

Reply:
> I don't get it, if we are defining that request is of type Request above, doesn't that ensure that it won't be null (or any other type)?

Even if TypeScript guarantees the type during compile-time, null or undefined values can still occur during runtime, hence we use safety checks as an extra layer of protection against potential null or undefined references in JavaScript runtime.

Have in mind that 90% of the codebase is still written in JS, so we are never really sure what values are passed to these functions, it's much better to be type safe than to create regressions in the long term. 

> Also, how/when will we remove all the unneeded safety checks?

Once majority of the codebase is migrated we can start removing these optional and null values and add run time validation (example [zod](https://github.com/colinhacks/zod) library). It is up to us when we do it, but now we have to keep migrating with excessive safety in mind.

Reply:
Got it, makes sense. Thanks for all that context!